File: docs/src/test/java/jdocs/home/cleanup/CleanupDocExample.java
/*
 * Copyright (C) 2022 - 2025 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.home.cleanup;

import akka.actor.typed.ActorSystem;
import akka.persistence.query.PersistenceQuery;
import akka.persistence.query.javadsl.CurrentPersistenceIdsQuery;
import akka.persistence.r2dbc.cleanup.scaladsl.EventSourcedCleanup;
import akka.persistence.r2dbc.query.javadsl.R2dbcReadJournal;
import scala.jdk.javaapi.FutureConverters;

public class CleanupDocExample {

  public static void example() {

    ActorSystem<?> system = null;

    // #cleanup
    CurrentPersistenceIdsQuery queries =
        PersistenceQuery.get(system)
            .getReadJournalFor(CurrentPersistenceIdsQuery.class, R2dbcReadJournal.Identifier());
    EventSourcedCleanup cleanup = new EventSourcedCleanup(system);

    int persistenceIdParallelism = 10;

    // forall persistence ids, delete all events before the snapshot
    queries
        .currentPersistenceIds()
        .mapAsync(
            persistenceIdParallelism,
            pid -> FutureConverters.asJava(cleanup.cleanupBeforeSnapshot(pid)))
        .run(system);

    // #cleanup

  }
}

File: docs/src/test/scala/docs/home/cleanup/CleanupDocExample.scala
/*
 * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.home.cleanup

import akka.actor.typed.ActorSystem
//#cleanup
import akka.persistence.query.PersistenceQuery
import akka.persistence.query.scaladsl.CurrentPersistenceIdsQuery
import akka.persistence.r2dbc.cleanup.scaladsl.EventSourcedCleanup
import akka.persistence.r2dbc.query.scaladsl.R2dbcReadJournal

//#cleanup

object CleanupDocExample {

  implicit val system: ActorSystem[_] = ???

  //#cleanup
  val queries = PersistenceQuery(system).readJournalFor[CurrentPersistenceIdsQuery](R2dbcReadJournal.Identifier)
  val cleanup = new EventSourcedCleanup(system)

  //  how many persistence ids to operate on in parallel
  val persistenceIdParallelism = 10

  // forall persistence ids, delete all events before the snapshot
  queries
    .currentPersistenceIds()
    .mapAsync(persistenceIdParallelism)(pid => cleanup.cleanupBeforeSnapshot(pid))
    .run()

  //#cleanup

}
